src/main/java/cucumber/runtime/formatter/SerenityReporter.java

```diff
@@ -48,50 +48,24 @@
  */
 public class SerenityReporter implements  Plugin,ConcurrentEventListener {
```

Comment:
Might be even easier to just change this to implement EventListener instead of ConcurrentEventListener, as - if I understood it correctly, for the 'old' EventListeners Cucumber4 will collect all events and then sends them in the correct order